pkg/http/server.go

```diff
@@ -123,9 +125,14 @@ func (srv *Server) Shutdown() error {
 		return errors.New("http server already in shutdown")
 	}
 
+	if atomic.LoadInt32(&srv.upAndRunning) <= 0 {
+		return nil
+	}
+
 	// Close underneath HTTP listener.
 	srv.listenerMutex.Lock()
 	err := srv.listener.Close()
```

Comment:
You could check `srv.listener == nil` than bringing additional field to the structure.